contracts/ligo/src/english_auction/english_auction_tez_permit.mligo

```diff
@@ -37,10 +36,10 @@ let config_storage_with_permit (p, permit_storage : permit_config_param * permit
     | None -> 
       let auction_storage = configure_auction_storage(p.config, Tezos.sender, permit_storage.auction_storage) in
       {permit_storage with auction_storage = auction_storage; counter = permit_storage.counter + 1n}
-    | Some permit -> 
-        let unsigned : bytes = ([%Michelson ({| { SELF; ADDRESS; CHAIN_ID; PAIR; PAIR; PACK } |} : nat * bytes -> bytes)] (permit_storage.counter, permit.paramHash) : bytes) in
-        assert_msg (Crypto.check permit.signerKey permit.signature unsigned, "INVALID_PERMIT"); 
-        assert_msg (Crypto.blake2b (Bytes.pack p.config) = permit.paramHash, "Config params and param hash do not match");
+    | Some permit ->
+        let param_hash = Crypto.blake2b (Bytes.pack p.config) in
+        let unsigned : bytes = ([%Michelson ({| { SELF; ADDRESS; CHAIN_ID; PAIR; PAIR; PACK } |} : nat * bytes -> bytes)] (permit_storage.counter, param_hash) : bytes) in
+        assert_msg (Crypto.check permit.signerKey permit.signature unsigned, "MISSIGNED"); 
         
         let configure_param = p.config in
```

Comment:
`configure_param` is not used in code.
Also both cases duplicate code which `configure_auction_storage` and return updated `permit_storage`.
Consider changing `match` to return seller address and then apply common logic:

```ocaml
const seller = match match p.optional_permit with
  | None -> Tezos.sender
  | Some permit ->
    ...
    address_from_key permit.signerKey
 in
 let auction_storage = configure_auction_storage(p.config, seller, permit_storage.auction_storage) in
 {permit_storage with auction_storage = auction_storage; counter = permit_storage.counter + 1n} 
```

Reply:
fixed thanks